src/main/haskell/kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -61,20 +67,55 @@ symbolVerifiers =
       , Builtin.verifySymbol Bool.assertSort [trivialVerifier, trivialVerifier])
     , ("KEQUAL.neq"
       , Builtin.verifySymbol Bool.assertSort [trivialVerifier, trivialVerifier])
+    , ("KEQUAL.ite", iteVerifier)
     ]
   where
     trivialVerifier :: Builtin.SortVerifier
     trivialVerifier = const $ const $ Right ()
 
-{- | @builtinFunctions@ defines the hooks for @KEQUAL.eq@ and @KEQUAL.neq@
-which can take arbitrary terms (of the same sort) and check whether they are
-equal or not, producing a builtin boolean value.
+    iteVerifier :: Builtin.SymbolVerifier
+    iteVerifier
+        findSort
+        SentenceSymbol
+            { sentenceSymbolSorts = sorts
+            , sentenceSymbolResultSort = result
+            }
+      =
+        Kore.Error.withContext "In argument sorts" $ do
+            Kore.Error.koreFailWhen
+                (arity /= 3)
+                ("Expected 3 arguments, found " ++ show arity)
+            Bool.assertSort findSort firstSort
+            Kore.Error.koreFailWhen
+                (secondSort /= thirdSort)
+                "Expected continuations to match"
+            Kore.Error.koreFailWhen
+                (secondSort /= result)
+                "Expected continuations to match"
+            return ()
+      where
+        arity = length sorts
+        firstSort = head sorts
```

Comment:
You can avoid the (admittedly, guarded) partial functions using a pattern match:
```.hs
case sorts of
    [firstSort, secondSort, thirdSort] -> _
    _ -> Kore.Error.koreFail _
```

Reply:
Whoops, I was going to make this prettier, but I forgot after all the debugging. Thanks!